test/framework/workload.go

```diff
@@ -21,3 +32,60 @@ func (w *WorkloadCluster) DeleteCluster(opts ...CommandOpt) {
 	opts = append(opts, withKubeconfig(w.ManagementClusterKubeconfigFile()))
 	w.deleteCluster(opts...)
 }
+
+// ApplyClusterManifest uses client-side logic to create/update objects defined in a cluster yaml manifest.
+func (w *WorkloadCluster) ApplyClusterManifest() {
+	ctx := context.Background()
+	w.T.Logf("Applying workload cluster %s spec located at %s", w.ClusterName, w.ClusterConfigLocation)
+	w.applyClusterManifest(ctx)
+	w.waitForKubeconfig(ctx)
+	w.ValidateClusterState()
+	w.StopIfFailed()
+}
+
+// DeleteClusterWithKubectl uses client-side logic to delete a cluster.
+func (w *WorkloadCluster) DeleteClusterWithKubectl() {
+	ctx := context.Background()
+	w.T.Logf("Deleting workload cluster %s with kubectl", w.ClusterName)
+	if err := w.KubectlClient.DeleteCluster(ctx, w.managementCluster(), w.cluster()); err != nil {
+		w.T.Fatalf("failed to delete workload cluster config: %s", err)
+	}
+	w.ValidateClusterDelete()
+	w.StopIfFailed()
+}
+
+func (w *WorkloadCluster) applyClusterManifest(ctx context.Context) {
+	if err := w.KubectlClient.ApplyManifest(ctx, w.ManagementClusterKubeconfigFile(), w.ClusterConfigLocation); err != nil {
+		w.T.Fatalf("Failed to apply workload cluster config: %s", err)
+	}
+}
+
+func (w *WorkloadCluster) waitForKubeconfig(ctx context.Context) {
+	w.T.Logf("Waiting for workload cluster %s kubeconfig to be available", w.ClusterName)
+	err := retrier.Retry(12, 5*time.Second, func() error {
+		return w.writeKubeconfigToDisk(ctx)
+	})
+	if err != nil {
+		w.T.Fatalf("Failed waiting for cluster kubeconfig: %s", err)
+	}
+}
+
+func (w *WorkloadCluster) writeKubeconfigToDisk(ctx context.Context) error {
+	secret, err := w.KubectlClient.GetSecretFromNamespace(ctx, w.ManagementClusterKubeconfigFile(), fmt.Sprintf("%s-kubeconfig", w.ClusterName), constants.EksaSystemNamespace)
```

Comment:
do we have a method somewhere to get the kubeconfig secret name? maybe even upstream capi has one we can use.